File: ruby/skill.rb
#!/usr/bin/env ruby 

a = 1
b = 'foo'
c = %w(foo bar bazz)


def sample(a, b, c)
  puts a
  puts b
  puts c

  a = 2
  b = 'bar'
  c = %w(hoge fuga piyo)

  puts 'a is : ' + a.to_s
  puts 'b is : ' + b.to_s
  puts 'c is : ' + c.to_s

end

def sample2(a, b, c)
  puts a
  puts b
  puts c

  a = 2
  b << 'bar'
  c.concat  %w(hoge fuga piyo)

  puts 'a is : ' + a.to_s
  puts 'b is : ' + b.to_s
  puts 'c is : ' + c.to_s

end


#sample a, b, c
sample2 a, b, c

puts 'a is : ' + a.to_s
puts 'b is : ' + b.to_s
puts 'c is : ' + c.to_s

File: js/skill.js
#!/usr/bin/env node 

var a = 1
var b = 'foo'
var c = ['foo', 'bar', 'bazz']

sample(a, b, c)
sample2(a, b, c)

console.log('a is : ' + a);
console.log('b is : ' + b);
console.log('c is : ' + c);

function sample(a, b, c) {

    a = 2
    b = 'hoge'
    c = ['hoge', 'fuga', 'piyo']

    console.log('a is : ' + a);
    console.log('b is : ' + b);
    console.log('c is : ' + c);
}

function sample2(a, b, c) {

    a = 2
    b = 'hoge'
    c.push(['hoge', 'fuga', 'piyo'])

    console.log('a is : ' + a);
    console.log('b is : ' + b);
    console.log('c is : ' + c);
}